kong/plugins/aws-lambda/handler.lua

```diff
@@ -240,6 +241,16 @@ function AWSLambdaHandler:access(conf)
     headers[VIA_HEADER] = VIA_HEADER_VALUE
   end
 
+  -- TODO: remove this in the next major release
+  -- function to remove array_mt metatables from empty tables
+  -- This is just a backward compatibility code to keep a
+  -- long-lived behavior that Kong responsed JSON objects
+  -- instead of JSON arrays for empty arrays.
+  local ct = headers["Content-Type"]
+  if ct and ct:lower():match("application/.*json") and conf.empty_arrays_mode == "legacy" then
+    content = remove_array_mt_for_empty_table(content)
+  end
+
```

Comment:
Faster checks first.

```suggestion
  if conf.empty_arrays_mode == "legacy" then
    local ct = headers["Content-Type"]
    if ct and ct:lower():match("application/.*json") and  then
      content = remove_array_mt_for_empty_table(content)
    end
  end
```

Reply:
Where does this `ct:lower():match("application/.*json")` come from, does it come directly from AWS-library?

Reply:
Yes, it does

Fixed